src/components/reservation/reservation-modal.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from "react";
+import Various from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/Various.svg";
+import {
+  ManifestationsSimpleFieldsFragment,
+  WorkMediumFragment
+} from "../../core/dbc-gateway/generated/graphql";
+import {
+  useAddReservationsV2,
+  useGetBranches,
+  useGetHoldingsV3,
+  useGetPatronInformationByPatronIdV2
+} from "../../core/fbs/fbs";
+import {
+  convertPostIdToFaustId,
+  creatorsToString,
+  filterCreators,
+  flattenCreators
+} from "../../core/utils/helpers/general";
+import Modal from "../../core/utils/modal";
+import { useText } from "../../core/utils/text";
+import { FaustId, Pid } from "../../core/utils/types/ids";
+import { Button } from "../Buttons/Button";
+import { Cover } from "../cover/cover";
+import ReservationFormListItem from "./ReservationFormListItem";
+import {
+  AgencyBranch,
+  AuthenticatedPatronV6,
+  HoldingsForBibliographicalRecordV3
+} from "../../core/fbs/model";
+import UserListItems from "./UserListItems";
+import ReservationSucces from "./ReservationSucces";
+import ReservationError from "./ReservationError";
+import { getPreferredLocation } from "../../apps/material/helper";
+
+export const reservationModalId = (faustId: FaustId) =>
+  `reservation-modal-${faustId}`;
+
+type ReservationModalProps = {
+  manifestation: ManifestationsSimpleFieldsFragment;
+  work: WorkMediumFragment;
+};
+
+const ReservationModal = ({ manifestation }: ReservationModalProps) => {
+  const {
+    pid,
+    materialTypes,
+    creators,
+    titles,
+    publicationYear,
+    edition: { summary }
+  } = manifestation;
+  const [reservationDidSuccess, setReservationDidSuccess] = useState<
+    boolean | null
+  >(null);
+  const t = useText();
+  const faustId = convertPostIdToFaustId(pid as Pid);
+  const { mutate } = useAddReservationsV2();
+
+  const branchResponse = useGetBranches();
+  const userResponse = useGetPatronInformationByPatronIdV2();
+  const holdingsResponse = useGetHoldingsV3({
+    recordid: [String(faustId)]
+  });
+
+  if (
+    !faustId ||
+    !branchResponse.data ||
+    !userResponse.data ||
+    !holdingsResponse.data
+  ) {
+    return null;
+  }
+
+  const { data: branchData } = branchResponse as { data: AgencyBranch[] };
+  const { data: userData } = userResponse as { data: AuthenticatedPatronV6 };
+  const { data: holdingsData } = holdingsResponse as {
+    data: HoldingsForBibliographicalRecordV3[];
+  };
+  const { reservations, holdings } = holdingsData[0];
+  const { patron } = userData;
+
+  // TODO move to material/helper.ts because it is used in multiple places (Info text under buttons in the material header)
+  const totalReservations = reservations;
```

Comment:
Consider whether the `totalReservations` variable is needed.

As far as I can tell `reservations`as not used elsewhere.